### DISCUSSION TITLE
Where do AOVs end up?

### DISCUSSION THREAD

**Discussion by Vertexwahn** [mitsuba-renderer/mitsuba3#435] [Q&A] [answered]
I wonder what happens with AOVs - i.e. where do they end up. When I look at `integrator.cpp` (line: 270) I see there is some memory reserved `std::unique_ptr<Float[]> aovs(new Float[n_channels]);`.
That is then handed down:
`render -> render_block(aovs) -> render_sample(aovs) -> block->put(aovs)`

It seems finally 
`block->put(box_filter ? pos : sample_pos, aovs, active);`
puts it via `void put(const Point2f &pos, const Float *values, Mask active = true);` to an ImageBlock.

Is this correct? Does the ImageBlock also hold the AOV values? For example normals, depth, albedo, shape index, etc.?
Are the AOVs filtered the same way as the "normal picture"?

**Comment by njroussel** [comment, marked as answer]
Hi @Vertexwahn 

Yes, they also end up in the `ImageBlock` and the `put` method you point out takes care of applying the `ReconstructionFilter` on all channels (aovs included).

**Reply by njroussel** [reply]
For completeness, let me add the following:

The AOVs will then eventually, in most use cases, make their way into a multi-channel `Bitmap` object. The following tutorial has a few examples of how such an Object behaves in Python and even a few examples specifically for AOVs.
